File: 2020/day2/part2/main_test.go
package part2

import (
	"advent-of-code/advent_of_code/day2/part1"
	"advent-of-code/advent_of_code/day2/part2"
	"testing"
)

func TestIsPolicyValid(t *testing.T) {
	p1 := part1.Policy{1, 4, "a", "abcde"}
	if true != part2.IsPolicyValid(p1) {
		t.Errorf("Mismatch! Expected %t, got %t", true, part2.IsPolicyValid(p1))
	}
	p2 := part1.Policy{1, 4, "b", "cdefg"}
	if false != part2.IsPolicyValid(p2) {
		t.Errorf("Mismatch! Expected %t, got %t", false, part2.IsPolicyValid(p2))
	}
	p3 := part1.Policy{1, 4, "c", "ccccccccc"}
	if false != part2.IsPolicyValid(p2) {
		t.Errorf("Mismatch! Expected %t, got %t", false, part2.IsPolicyValid(p3))
	}
}

func TestMain(t *testing.T) {
	input_data := part2.Main("../input_test.txt")
	if input_data != 1 {
		t.Errorf("Mismatch! Expected %d, got %d", 2, input_data)
	}
}

File: 2020/day2/part2/main_test.py
import unittest

from advent_of_code.day2.part2.main import Policy
from advent_of_code.day2.part2.main import is_password_valid, main
class TestDay2Part2(unittest.TestCase):

    def test_is_policy_valid(self):
        p1: Policy = Policy(*(1, 3, "a", "abcde"))
        self.assertTrue(is_password_valid(p1))
        p2: Policy = Policy(*(1, 3, "b", "cdefg"))
        self.assertFalse(is_password_valid(p2))
        p3: Policy = Policy(*(2, 9, "c", "ccccccccc"))
        self.assertFalse(is_password_valid(p3))

    def test_main(self):
        input_data: int = main("tests/day2/input_test.txt")
        self.assertEqual(input_data, 1)
    